subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c

```diff
@@ -925,6 +925,8 @@ static inline int isr_rx_pdu(struct lll_scan *lll, struct pdu_adv *pdu_adv_rx,
 		ret = lll_stop(lll);
```

Comment:
Could you please analyze the use of `lll_stop`, this interface along the `lll_is_stop` used in `prepare_cb` should handle the race (unless its implementation is flawed) and bypass any subsequent connection indication being transmitted.

Reply:
Thanks for the pointer, I think I could make `lll_stop` also work for our LLL. However, I wonder whether the per-LLL stop flag would also work with coded PHY. As far as I understand the code in ll_create_connection(), the normal scanner and the coded PHY scanner point to the same connection object and could race each other?

Reply:
> the normal scanner and the coded PHY scanner point to the same connection object and could race each other?

Yes, good you point out. This is not addressed today as the initiator for extended advertising is not yet implemented.

Could we use the `lll_stop` in the 1M and CODED_PHY scanning to stop each other's LLL?

Reply:
I think it would be possible to have the LLLs stop each other, but I am not sure whether that is the best solution. As far as I can tell, the current uses of `lll_stop` are all related to the start of a connection and could all be replaced with a flag in the `lll_conn` struct. That would mean that we can remove the `is_stop` flag from the general `lll_hdr` and only include a respective flag in `lll_conn`. Sharing of a connection object between LLLs would be taken care of automatically. Should I give that a shot or do you prefer sticking to `lll_stop`?

Reply:
> Should I give that a shot or do you prefer sticking to lll_stop?

I have a hunch, I will not stick to `lll_stop`. I dont remember now, but I have the feeling use of `lll_stop` for some scenario did not work.

I think this was during testing: https://github.com/zephyrproject-rtos/zephyr/pull/30735
Possibly when central was trying to connect to device as fast as it could. But I am not able remember it to document a symptom or root cause here (now).

Hence, please go ahead with your solutions to the problem at hand here.

My change request here is, could the scanner context hold the flag instead of each of n connection context? (I know, the flag is in an union, but to future-proof and had the flag to the context)

Reply:
I have reworked the change now to use a more general "initiated" flag in the connection object and remove the use of lll_stop.